components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/AnalyticsPublisher.java

```diff
@@ -0,0 +1,303 @@
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.bpmn.analytics.publisher.internal.BPMNAnalyticsHolder;
+import org.wso2.carbon.bpmn.analytics.publisher.models.BPMNProcessInstance;
+import org.wso2.carbon.bpmn.analytics.publisher.models.BPMNTaskInstance;
+import org.wso2.carbon.bpmn.analytics.publisher.utils.AnalyticsPublishServiceUtils;
+import org.wso2.carbon.bpmn.analytics.publisher.utils.BPMNDataReceiverConfig;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.context.RegistryType;
+import org.wso2.carbon.databridge.agent.thrift.DataPublisher;
+import org.wso2.carbon.databridge.agent.thrift.exception.AgentException;
+import org.wso2.carbon.databridge.commons.AttributeType;
+import org.wso2.carbon.databridge.commons.StreamDefinition;
+import org.wso2.carbon.databridge.commons.exception.*;
+import org.wso2.carbon.registry.api.Registry;
+import org.wso2.carbon.registry.api.RegistryException;
+import org.wso2.carbon.registry.core.utils.RegistryUtils;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import java.net.MalformedURLException;
+import java.util.concurrent.Executors;
+
+/**
+ * AnalyticsPublisher uses to publish events to the data receiver in data-bridge
+ */
+public class AnalyticsPublisher {
+    private static final Log log = LogFactory.getLog(AnalyticsPublisher.class);
+
+    private String processInstanceStreamId;
+    private String taskInstanceStreamId;
+    private DataPublisher dataPublisher;
+    private AnalyticsPublishServiceUtils analyticsPublishServiceUtils;
+
+    /**
+     * Initialize the objects for AnalyticsPublisher
+     */
+    public void initialize() {
+        try {
+            RegistryUtils.setTrustStoreSystemProperties();
+            dataPublisher = createDataPublisher();
+            if(dataPublisher != null){
+                processInstanceStreamId = getBPMNProcessInstanceStreamId();
+                taskInstanceStreamId = getBPMNTaskInstanceStreamId();
+                analyticsPublishServiceUtils = new AnalyticsPublishServiceUtils();
+                int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+                String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+                Registry registry = BPMNAnalyticsHolder.getInstance().getRegistryService()
+                        .getGovernanceSystemRegistry();
+                if (processInstanceStreamId == null && taskInstanceStreamId == null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Stream definitions are null...");
+                    }
+                } else {
+                    setPrivilegeContext(tenantId, tenantDomain, registry);
+                }
+            }else{
+             if(log.isDebugEnabled()){
+                 log.debug("Data Publisher object is null...");
```

Comment:
Those formatting issues were already fixed and you can see them under other  pull requests.